codex-cli/src/components/chat/terminal-chat.tsx

```diff
@@ -275,6 +278,49 @@ export default function TerminalChat({
     };
   }, [loading, confirmationPrompt]);
 
+  // Notify desktop with a preview when an assistant response arrives
+  const prevLoadingRef = useRef<boolean>(false);
+  useEffect(() => {
```

Comment:
First, we should carve out a general notification abstraction rather than duping this logic into the middle of a UI component.